src/support/socket.h

```diff
@@ -68,6 +68,28 @@ static inline int poll(struct pollfd* pfd, int nfds, int timeout) {
 
 namespace tvm {
 namespace support {
+
+/*!
+ * \brief Call a function and retry if an EINTR error is encountered.
+ * \param f The function to retry.
+ * \param error_value The error value returned by the call on retry failure.
+ * \return The return code returned by function f or error_value on retry failure.
+ */
+template <typename T>
+T retry_call(std::function<T()> f, T error_value) {
```

Comment:
Prefer not using std::function, because that means construction of std::function in cases where it can be inlined. Instead, use the following signature so f can be inlined.

```c++
template <typename F>
T RetryCallWhenEINTR(F f, T error_value);
```